scripts/firedrake-install

```diff
@@ -1599,7 +1617,10 @@ else:
         petsc_changed = git_update("petsc", deps["petsc"])
     else:
         petsc_changed = False
-    petsc4py_changed = git_update("petsc4py", deps["petsc4py"])
+    if use_petsc4py_in_petsc:
+        petsc4py_changed = petsc_changed
```

Comment:
In upgrade mode, `use_petsc4py_in_petsc` is always true. so no branch required.

Reply:
Thanks! Removed `petsc4py_changed` variable, as `petsc_changed` will do.

Reply:
Actually, `petsc4py_changed` is now used as a flag to indicate that we have moved `src/petsc4py` -> `src/petsc4py_old`, as whenever this happens, we need to reinstall `petsc4py`.